src/python/pants/core/goals/fmt.py

```diff
@@ -44,7 +44,7 @@ def skip(cls: type[_F], *, formatter_name: str) -> _F:
     @classmethod
     def from_process_result(
         cls,
-        process_result: ProcessResult,
+        process_result: ProcessResult | FallibleProcessResult,
```

Comment:
FYI they do have the option `----exit-zero-even-if-changed`, which would allow us to avoid this change. But I think this is fine to do.

Reply:
I actually prefer that, because it will let the non-zero exit be used elsewhere (since we are putting the exit code into the metadata) it other tools (like buildsnese) to know that the tool behaved in a certain way.

Reply:
Okay cool! Then ignore my diff, the `argv` will be different whether it's `fmt` vs. `lint`. Instead, add this to `SetupRequest`:

https://github.com/pantsbuild/pants/blob/b1d7ab9d245c5fb6616477c6e448353d5ac5a7fc/src/python/pants/backend/python/lint/docformatter/rules.py#L40-L43

and update accordingly